File: app/src/main/java/com/cooper/wheellog/views/ViewBlockInfo.java
package com.cooper.wheellog.views;

import java.util.concurrent.Callable;

public class ViewBlockInfo implements Comparable {
    private String title;
    private boolean enabled;
    private Callable<String> value;
    private int index = -1;

    public ViewBlockInfo(String title, Callable<String> value, boolean enabled) {
        this.title = title;
        this.value = value;
        this.enabled = enabled;
    }

    public ViewBlockInfo(String title, Callable<String> value) {
        this.title = title;
        this.value = value;
        this.enabled = true;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getValue() throws Exception {
        return value.call();
    }

    public void setValue(Callable<String> mValue) {
        this.value = mValue;
    }

    public boolean getEnabled() {
        return this.enabled;
    }

    public void setEnabled(boolean enabled) {
        this.enabled = enabled;
    }

    public int getIndex() {
        return this.index;
    }

    public void setIndex(int value) {
        index = value;
    }

    @Override
    public int compareTo(Object o) {
        return Integer.compare(this.index, ((ViewBlockInfo)o).getIndex());
    }
}

File: app/src/main/java/com/cooper/wheellog/views/ViewBlockInfo.kt
package com.cooper.wheellog.views

import java.util.concurrent.Callable

class ViewBlockInfo : Comparable<ViewBlockInfo> {
    var title: String
    var enabled: Boolean
    private var value: Callable<String>
    var index = -1

    constructor(title: String, value: Callable<String>, enabled: Boolean) {
        this.title = title
        this.value = value
        this.enabled = enabled
    }

    constructor(title: String, value: Callable<String>) : this(title, value, true)

    @Throws(Exception::class)
    fun getValue(): String {
        return value.call()
    }

    fun setValue(mValue: Callable<String>) {
        value = mValue
    }

    override fun compareTo(other: ViewBlockInfo): Int {
        return index.compareTo(other.index)
    }
}